Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,760 @@
+            if (parent)
+            {
+                d_positions.setParent(parent);
+                d_positions.setReadOnly(true);
+            }else{
+                msg_warning(this) << "No attribute 'rest_position' in component '" << getName() << "'.\n"
+                                  << "The BaseROI component thus have no input and is thus deactivated.\n" ;
+                d_componentState.setValue(ComponentState::Invalid) ;
+                return ;
+            }
+        }
+        else
+        {
+            MeshLoader* loader = nullptr ;
+            this->getContext()->get(loader, core::objectmodel::BaseContext::Local);
+            if (loader)
+            {
+                BaseData* parent = loader->findData("position");
+                if (parent)
+                {
+                    d_positions.setParent(parent);
+                    d_positions.setReadOnly(true);
+                }else{
+                    msg_warning(this) << "No attribute 'position' in component '" << getName() << "'.\n"
+                                      << "The BaseROI component thus have no input and is thus deactivated.\n" ;
+                    d_componentState.setValue(ComponentState::Invalid) ;
+                    return ;
+                }
+            }
+            else   // no local state, no loader => find upward
+            {
+                this->getContext()->get(mstate, core::objectmodel::BaseContext::SearchUp);
+                if(!mstate){
+                    msg_error(this) <<  "Unable to find a MechanicalObject for this component. "
+                                        "To remove this error message you can either:\n"
+                                        "   - to specifiy the DOF where to apply the BaseROI with the 'position' attribute.\n"
+                                        "   - to add MechanicalObject or MeshLoader component before the BaseROI in the scene graph.\n";
+                    d_componentState.setValue(ComponentState::Invalid) ;
+                    return ;
+                }
+
+                BaseData* parent = mstate->findData("rest_position");
+                if(!parent){
+                    dmsg_error(this) <<  "Unable to find a rest_position attribute in the MechanicalObject '" << mstate->getName() << "'";
+                    d_componentState.setValue(ComponentState::Invalid) ;
+                    return ;
+                }
+                d_positions.setParent(parent);
+                d_positions.setReadOnly(true);
+            }
+        }
+    }
+
+
+    if (!d_edges.isSet() || !d_triangles.isSet() || !d_tetrahedra.isSet() || !d_hexahedra.isSet() || !d_quads.isSet() )
+    {
+        msg_info(this) << "No topology given. Searching for a TopologyContainer and a BaseMeshTopology in the current context.\n";
+
+        TopologyContainer* topologyContainer;
+        this->getContext()->get(topologyContainer, core::objectmodel::BaseContext::Local);
```

Comment:
`TopologyContainer` derives from `BaseMeshTopology`, so I think if you get a `BaseTopology`, you necessarily get a `TopologyContainer`. Therefore, I am not sure that looking for a `TopologyContainer` is pertinent.

Reply:
I guess you meant `so I think if you get a TopologyContainer, you necessarily get a BaseTopology` right ?
In that case I would agree.

Reply:
Haha @alxbilger, you put the finger in a real deep crap.
There are some tests/scenes where there are two topologies (e.g a grid and a topologyContainer). So after removing the topologycontainer search, the get() finds the grid and not the topologyContainer... and why not, after all... 🤡

(see https://github.com/sofa-framework/sofa/blob/master/Sofa/Component/ODESolver/Backward/tests/StaticSolver_test.cpp)

This is so wrong 💩 one day, we should enforce explicit links 👮

Reply:
ok I see... 🤯